infra/resources/dev/function_app.tf

```diff
@@ -38,6 +39,33 @@ module "function_app" {
   tags = local.tags
 }
 
+module "azure_function_v3_function_app" {
+  source  = "pagopa-dx/azure-function-app/azurerm"
+  version = "~> 0"
+
+  environment         = merge(local.environment, { app_name = "v3" })
+  tier                = "s"
+  resource_group_name = data.azurerm_resource_group.test_rg.name
+
+  virtual_network = {
+    name                = data.azurerm_virtual_network.test_vnet.name
+    resource_group_name = data.azurerm_virtual_network.test_vnet.resource_group_name
+  }
+
+  subnet_pep_id = data.azurerm_subnet.pep_snet.id
+  subnet_cidr   = "10.50.8.0/24"
```

Comment:
idea: can you try the new method from the azure provider?

https://registry.terraform.io/providers/pagopa-dx/azure/latest/docs/resources/available_subnet_cidr

Reply:
Thanks for the suggestion, it's a good idea!